model/pipeline/src/main/proto/org/apache/beam/model/pipeline/v1/beam_runner_api.proto

```diff
@@ -1590,6 +1592,10 @@ message ExternalPayload {
   map<string, string> params = 2;  // Arbitrary extra parameters to pass
 }
 
+message AnyOfEnvironmentPayload {
+  repeated Environment environments = 1;
```

Comment:
I assume these are intended to be any arbitrarily fully specified environment in this list. 

That is, they won't implicitly depend on anything at the "root" environment?

Reply:
Correct. Clarified.